docs/dls/component-design-guidelines.md

```diff
@@ -0,0 +1,100 @@
+---
+id: component-design-guidelines
+title: Component Design Guidelines
+description: Documentation on Design
+---
+
+Be it a new component contribution, or plugin specific components, you'll want
+to follow these guidelines, we'll cover the 3 main subjects that define the
+general look and feel of your components, all of which build on top of the MUI
+theme features:
+
+- Layout
+- Color palette
+- Typography
+
+## 🏗️ Layout
+
+Layout refers to how you organize or stack content, whenever possible we want
+to use Backstage's components (check the [Storybook][1] for a list and demo)
+or MUI components (check the [MUI docs][2]).
+
+If none of these fit your layout needs, then you can build one, however using
+HTML+CSS directly is not recommended, it's better to use MUI layout components
+to make your layout theme aware, meaning if someone changes the theme, your
+layout would react to those changes without requiring updates to your code.
+
+Specifically you want to look at these components that make use of the
+`theme.spacing()` function for margins, paddings and positions, as well as
+color palette and typography:
+
+- [Container][3] mostly at page level
+- [Box][4] like a div that can be customized a lot
+- [Grid][5] & [Grid V2][6] (preferable V2) for flexible grid layouts
```

Comment:
I think you'll want to look through these links and all of the other ones below too. We are on MUI4, so all of the links should be to the version 4 docs and at least Grid2 isn't there at all.

Reply:
I didn't want to put the V4 explicitly in case we update Backstage to latest and we forget to update the docs here, in this form it would always point to latest MUI, should I change it regardless?

Reply:
Changed to V4